qemu/tests/block_resize.py

```diff
@@ -16,156 +175,76 @@ def run(test, params, env):
-                                                                timeout)
+    pattern = params.get("block_size_pattern")
+    block_size_cmd = params["block_size_cmd"]
+    accept_ratio = float(params.get("accept_ratio", 0))
     data_image = params.get("images").split()[-1]
     data_image_params = params.object_params(data_image)
-    data_image_size = data_image_params.get("image_size")
-    data_image_size = float(utils_misc.normalize_data_size(data_image_size,
-                                                           order_magnitude="B"))
     data_image_filename = storage.get_image_filename(data_image_params,
                                                      data_dir.get_data_dir())
     data_image_dev = vm.get_block({'file': data_image_filename})
-
+    block_virtual_size = vm.monitor.get_block_virtual_size({'device': data_image_dev})
+    disk_change_ratio = params.get("disk_change_ratio")
+    block_resize_type = params["block_resize_type"]
     drive_path = ""
-    if params.get("os_type") == 'linux':
-        pattern = r"(serial|wwn)=(\w+)"
-        match = re.search(pattern, params["blk_extra_params_%s"
-                          % data_image], re.M)
-        if match:
-            drive_id = match.group(2)
-        else:
-            test.fail("No available tag to get drive id")
-        drive_path = utils_misc.get_linux_drive_path(session, drive_id)
-        if not drive_path:
-            test.error("Failed to get '%s' drive path" % data_image)
-
-    block_size_cmd = params["block_size_cmd"].format(drive_path)
-    block_size_pattern = params.get("block_size_pattern")
-    need_reboot = params.get("need_reboot", "no") == "yes"
-    accept_ratio = float(params.get("accept_ratio", 0))
 
-    error_context.context("Check image size in guest", logging.info)
-    block_size = get_block_size(session, block_size_cmd, block_size_pattern)
-    if (block_size > data_image_size or
-            block_size < data_image_size * (1 - accept_ratio)):
-        test.error("Image size from guest and image not match"
-                   "Block size get from guest: %s \n"
-                   "Image size get from image: %s \n"
-                   % (block_size, data_image_size))
+    if params.get("os_type") == 'linux':
+        drive_path = get_drive_path_linux(test, session, params, data_image)
+        block_size_cmd = params["block_size_cmd"].format(drive_path)
 
-    if params.get("guest_prepare_cmd"):
-        session.cmd(params.get("guest_prepare_cmd"))
+    if params.get("os_type") == "windows" and driver_name:
+        session = utils_test.qemu.windrv_check_running_verifier(session, vm,
+                                                                test, driver_name,
+                                                                timeout)
+        if params.get("format_disk", "no") == "yes":
+            error_context.context("Format disk", logging.info)
+            utils_misc.format_windows_disk(session, params["disk_index"],
```

Comment:
the function \"format_windows_disk()\" are update in avocado-vt framework, please modify the arguments.

Reply:
updated, thanks~